### ISSUE TITLE
Core layer todo list for 2.2.X [NEED CONTRIBUTION]

### ISSUE THREAD

**Issue by zsdonghao** [tensorlayer/tensorlayer#956]
- Discuss in Slack [(invited link)](https://join.slack.com/t/tensorlayer/shared_invite/enQtMjUyMjczMzU2Njg4LWI0MWU0MDFkOWY2YjQ4YjVhMzI5M2VlZmE4YTNhNGY1NjZhMzUwMmQ2MTc0YWRjMjQzMjdjMTg2MWQ2ZWJhYzc)

- Docs
  - [ ] installation
  - [ ] layers
  - [ ] the links in https://github.com/tensorlayer/awesome-tensorlayer
  - [ ] https://tensorlayer.readthedocs.io/en/2.0.0/user/examples.html
  - [x] https://tensorlayer.readthedocs.io/en/latest/modules/layers.html#batch-normalization   batchnorm1d 2d 3d should be separated   Rundi
  - [x] others
- Layer
  - [x] Activation `act` supports string inputs e.g., `Dense(100, act='relu')` Ruihai
  - [ ] RNN `cell` supports string inputs e.g., `cell_fn='LSTM')`
  - [ ] Remove `as_layer()`
  - [ ] Quantized layers should be refactored
  - [ ] tl.optimizers should be refactored
  - [ ] Optimize the `build' of core layer 
  - [x] InstanceNorm 1d 2d 3d should be implemented  Yingda
  - [ ] automatically deal with `in_channels`:
           `super(MyLayer, self).__init__(name, in_channel=in_channel)`
  - [ ] refactor QuanConv2dWithBN and etc
  - [ ] Remove below codes in layers
```
        self.build()
        self._built = True
``` 
- Model
  - [x] Fix model core `_construct_graph` BFS.  @ChrisWu1997 
  - [x] Add callback features for specific layer output. @ChrisWu1997 

- Visualize
  - [ ] tl.vis.save_images  if input is a tensor, run x.numpy() automatically.
 - Examples (Tutorial)
  - [x] text_generation
  - [x] text_ptb
  - [ ] database [WIP] @ChrisWu1997 
  - [ ] distributed_training
  - [ ] [tutorial_binarynet_cifar10_tfrecord.py](https://github.com/tensorlayer/tensorlayer/blob/master/examples/quantized_net/tutorial_binarynet_cifar10_tfrecord.py)
  - [ ] [tutorial_binarynet_mnist_cnn.py](https://github.com/tensorlayer/tensorlayer/blob/master/examples/quantized_net/tutorial_binarynet_mnist_cnn.py)
  - [ ] [more in quantized_net](https://github.com/tensorlayer/tensorlayer/tree/master/examples/quantized_net)
- Initializer
  - [ ] [ours](https://tensorlayer.readthedocs.io/en/latest/modules/initializers.html) vs [Keras's](https://www.tensorflow.org/api_docs/python/tf/initializers)
- Other Examples
  - [ ] https://github.com/tensorlayer/srgan
  - [ ] https://github.com/tensorlayer/openpose-plus
  - [ ] https://github.com/tensorlayer/adaptive-style-transfer
  - [ ] https://github.com/tensorlayer/seq2seq-chatbot
  - [ ] more in https://github.com/tensorlayer/awesome-tensorlayer

- Next step
  - [ ] PointCNN
  - [ ] RL - ray/rllib
